x/concentrated-liquidity/position.go

```diff
@@ -222,30 +221,36 @@ func (k Keeper) deletePosition(ctx sdk.Context,
 	store := ctx.KVStore(k.storeKey)
 
 	// Remove the position ID to position mapping.
-	key := types.KeyPositionId(positionId)
-	if !store.Has(key) {
+	positionIdKey := types.KeyPositionId(positionId)
+	if !store.Has(positionIdKey) {
 		return types.PositionIdNotFoundError{PositionId: positionId}
 	}
-	store.Delete(key)
+	store.Delete(positionIdKey)
 
 	// Remove the address-pool-position ID to position mapping.
-	key = types.KeyAddressPoolIdPositionId(owner, poolId, positionId)
-	if !store.Has(key) {
+	addressPoolIdPositionIdKey := types.KeyAddressPoolIdPositionId(owner, poolId, positionId)
+	if !store.Has(addressPoolIdPositionIdKey) {
 		return types.AddressPoolPositionIdNotFoundError{Owner: owner.String(), PoolId: poolId, PositionId: positionId}
 	}
-	store.Delete(key)
+	store.Delete(addressPoolIdPositionIdKey)
 
 	// Remove the pool ID to position ID mapping.
-	key = types.KeyPoolPositionPositionId(poolId, positionId)
-	if !store.Has(key) {
+	poolIdKey := types.KeyPoolPositionPositionId(poolId, positionId)
+	if !store.Has(poolIdKey) {
 		return types.PoolPositionIdNotFoundError{PoolId: poolId, PositionId: positionId}
 	}
-	store.Delete(key)
+	store.Delete(poolIdKey)
 
 	// Remove the position ID to underlying lock ID mapping (if it exists)
-	key = types.KeyPositionIdForLock(positionId)
-	if store.Has(key) {
-		store.Delete(key)
+	positionIdLockKey := types.KeyPositionIdForLock(positionId)
+	if store.Has(positionIdLockKey) {
+		underlyingLockId, err := k.GetLockIdFromPositionId(ctx, positionId)
+		if err != nil {
+			return err
+		}
+		store.Delete(positionIdLockKey)
+		lockIdPositionKey := types.KeyLockIdForPositionId(underlyingLockId)
+		store.Delete(lockIdPositionKey)
```

Comment:
Do we have tests confirming the deletion /modification of both indexes? Applies to other CRUD methods

Also, as a sanity check, is import /export genesis handled correctly for both indexes?

Reply:
Extended the gotest to include checks before and after position deletion for lockId to positionId record https://github.com/osmosis-labs/osmosis/pull/4885/commits/04545e72285f1d99188fd7222667e71c650d2af3